chart/etcd-backup-restore/templates/etcd-statefulset.yaml

```diff
@@ -201,6 +201,13 @@ spec:
               name: {{ .Release.Name }}-etcd-backup
               key: "storageAPIEndpoint"
   {{- end }}
+  {{- if .Values.backup.gcs.emulatorEnabled }}
+        - name: "EMULATOR_ENABLED"
```

Comment:
Originally I prefixed it with `GOOGLE` like other params you mentioned. But I modified it because of the following [comments](https://github.com/gardener/etcd-backup-restore/pull/697#discussion_r1448325839). And later with this suggestion, Azurite PR got merged following the [same convention](https://github.com/gardener/etcd-backup-restore/blob/dc0f973b9db963b5a0042371ae3a125436f5801d/chart/etcd-backup-restore/templates/etcd-statefulset.yaml#L194-L199). That's why I have changed it as well to keep it same across the providers. WDYT?

Reply:
Ouch. It would have been better to use a provider-specific environment variable name to be very precise about which provider the emulator has been enabled for.
I would request you to discuss with @renormalize and see whether you can use the provider-specific prefix for these, as is done with the other provider environment variables.

Reply:
The initial convention we both had followed was similar to the one you suggested, but we changed the convention after @seshachalam-yv's [comment](https://github.com/gardener/etcd-backup-restore/pull/697#discussion_r1448325839). 

I have no strong opinions on this. I'll change the convention followed once all reach an agreement.

Reply:
Me & @shreyas-s-rao have come to agree upon the use of `PROVIDER_EMULATOR_ENABLED` format so that it is clear which emulator is being used for the job when we run etcdbr against multiple providers sequentially in pipelines, having a common env var causes ambiguities between deciding fake provider or actual provider across providers. 

All of this can be avoided by naming them with their prefix. I've made my changes in this respect. I would request @renormalize also to make similar change for the Azurite